pkg/controller/build/build_controller_test.go

```diff
@@ -113,21 +110,29 @@ var _ = Describe("Reconcile Build", func() {
 				}
 				buildSample.Spec.Output.SecretRef = nil
 
+				// Fake some client Get calls and ensure we populate all
+				// different resources we could get during reconciliation
+				client.GetCalls(func(context context.Context, nn types.NamespacedName, object runtime.Object) error {
```

Comment:
This get calls override the ones defined in the `BeforeEach` block, therefore you need to cover here the one when getting a `Build` in the Build controller reconciliation, otherwise you will hit some nil pointers which I believe made you add some validations in the reconciler around `build.spec.StrategyRef`

Reply:
As far as I know, before running each case, `BeforeEach` block will be run. That means fake Build object will be gotten each time. Why do we need to override Build within each case? (Sorry, maybe it's a simple question, but because I am unfamiliar with it, I want to clarify it more clearly.)